rosidl_generator_cpp/CMakeLists.txt

```diff
@@ -102,6 +102,11 @@ if(BUILD_TESTING)
       ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}
     )
   endif()
+  if(MSVC)
+    # https://developercommunity.visualstudio.com/content/problem/919371/c2017-illegal-escape-sequence-when-using-in-a-raw.html
+    set_source_files_properties(
+      test/test_traits.cpp PROPERTIES COMPILE_FLAGS "/experimental:preprocessor")
+  endif()
```

Comment:
Instead of relying on the pre-processor supporting raw strings, can we just use normal multi-line strings with the proper escape characters?
I understand that raw strings are much nicer, but I'm not what will happen when compiling in a system that doesn't support them (e.g.: an old windows VS).
If it's a way to detect if the system supports raw strings, and skip compiling the test in case if not I think it would be ok to use them.

Reply:
> Instead of relying on the pre-processor supporting raw strings, can we just use normal multi-line strings with the proper escape characters?

The test code intentionally uses a raw string literal so that the expected `yaml` can be copy-n-pasted into a parser / validator.

> I understand that raw strings are much nicer, but I'm not what will happen when compiling in a system that doesn't support them (e.g.: an old windows VS).
If it's a way to detect if the system supports raw strings, and skip compiling the test in case if not I think it would be ok to use them.

According to REP 2000 all active ROS 2 distros require at Visual Studio 2019. So I don't see that we need to support any older versions and therefore don't need any conditional logic.